crates/virtio-queue/src/chain.rs

```diff
@@ -88,6 +88,28 @@ where
         }
     }
 
+    /// Return a new instance of Writer
+    pub fn writer(self, mem: &'a GuestMemoryMmap) -> Result<Writer<'a>, Error>
+    where
+        M::Target: GuestMemory + Sized,
```

Comment:
Hm... Why is this needed here? M is not really used in this function, right?

Reply:
I can reduce it to:
```rust
    where
        M::Target: Sized,
```
But when removing the M I get:
```
the size for values of type `<M as Deref>::Target` cannot be known at compilation time
the trait `Sized` is not implemented for `<M as Deref>::Target`
chain.rs(94, 26): required by a bound introduced by this call
descriptor_utils.rs(310, 34): required by a bound in `Writer::<'a, B>::new`
chain.rs(92, 79): consider further restricting the associated type: ` where <M as Deref>::Target: Sized`
```